src/utils.cpp

```diff
@@ -452,6 +450,11 @@ class XorBinarize : public IdentityWalker
 
 // Given a boolean formula, removes terms "true" and "false" to give a formula
 // without adding new symbols
+
+// The way true and false are eliminated is by doing a preorder traversal. When
+// I am on a certain node, The children are already reducecd. Reduced means that
+// they are a term without true or false, or just true, or just false.
```

Comment:
and what do you do with just true and just false terms?

Reply:
If the entire formula is reduced to true or false, I return it. Should I do something else?

Reply:
But then how to you print the dimacs?

Reply:
If it is reduced to true/false. I will not pass it to print dimacs

Reply:
Sorry, I have handled the case of the term being true/false in the function cnf_to_dimacs

Reply:
True will be printed as 
: p cnf 0 0
False as:
p cnf 0 1
0